src/http_workers/worker_pool.rs

```diff
@@ -0,0 +1,35 @@
+use std::sync::Arc;
+
+use tokio::sync::{mpsc, oneshot, Mutex};
+
+use super::{
+    worker::{Worker, WorkerHandle},
+    Work,
+};
+
+pub struct WorkerPool<W>
+where
+    W: Work,
+{
+    receiver: mpsc::Receiver<oneshot::Sender<W>>,
+}
+
+impl<W> WorkerPool<W>
+where
+    W: Work + Send + Sync + Clone + 'static,
+{
+    pub async fn new(size: usize) -> WorkerPool<W> {
+        let (sender, receiver) = mpsc::channel(size);
+
+        for id in 0..size {
+            Worker::new(sender.clone()).await.run().await;
+        }
+
+        WorkerPool { receiver }
+    }
+
+    pub async fn recv(&mut self) -> WorkerHandle<W> {
```

Comment:
recv is a confusing name here. Because, as a user of the pool i have no idea what is the internal implementation of the pool or how it works. so Recv is confusing because it makes no sense from a Pool user prespective.

In general a Pool is a collection of certain type of resources (Connection Pool, Thread Pool) Hence, the main functionality of a Pool is to do a `get` operation to get one of the available free resources. 

so when you do `connection_pool.get()` you expect to get a connection. When you do ` worker_pool.get()` you get a worker. and so on.

```suggestion
    pub async fn get(&mut self) -> WorkerHandle<W> {
```